src/compiler/checker.ts

```diff
@@ -5047,8 +5016,25 @@ namespace ts {
         }
 
         function getTypeOfFuncClassEnumModule(symbol: Symbol): Type {
-            const links = getSymbolLinks(symbol);
+            let links = getSymbolLinks(symbol);
             if (!links.type) {
+                const jsDeclaration = getDeclarationOfJSInitializer(symbol.valueDeclaration);
+                if (jsDeclaration) {
+                    const jsSymbol = getSymbolOfNode(jsDeclaration);
+                    if (jsSymbol && (jsSymbol.exports && jsSymbol.exports.size || jsSymbol.members && jsSymbol.members.size)) {

```

Comment:
Style: Maybe we should add a `size` helper function that handles `undefined` like how we have `length` for arrays?

Reply:
Let me see how many times we check for null before accessing size. The level of inconvenience seems worth about 5-6 checks before making a helper.

Later: it's over 10 now. I'll add the helper.